src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.PackageDependencyResolution.targets

```diff
@@ -145,13 +145,27 @@ Copyright (c) .NET Foundation. All rights reserved.
 
   <UsingTask TaskName="Microsoft.NET.Build.Tasks.ResolvePackageDependencies"
              AssemblyFile="$(MicrosoftNETBuildTasksAssembly)" />
+  <UsingTask TaskName="Microsoft.NET.Build.Tasks.CheckForTargetInAssetsFile"
+           AssemblyFile="$(MicrosoftNETBuildTasksAssembly)" />
 
   <!-- The condition on this target causes it to be skipped during design-time builds if
         the restore operation hasn't run yet.  This is to avoid displaying an error in
         the Visual Studio error list when a project is created before NuGet restore has
         run and created the assets file. -->
   <Target Name="RunResolvePackageDependencies"
           Condition=" '$(DesignTimeBuild)' != 'true' Or Exists('$(ProjectAssetsFile)')">
+    
+    <!-- Verify that the assets file has a target for the right framework.  Otherwise, if we restored for the
+         wrong framework, we'd end up finding no references to pass to the compiler, and we'd get a ton of
+         compile errors.
+         
+         The RuntimeIdentifier isn't used to select references to pass to the compiler, so we don't include
+         it here. -->
+    <CheckForTargetInAssetsFile
+      AssetsFilePath="$(ProjectAssetsFile)"
+      TargetFrameworkMoniker="$(TargetFrameworkMoniker)"
+      RuntimeIdentifier="" />
```

Comment:
IIRC there often isn't a discernable difference between the values coming from the RIDfull and RIDless targets, which is why I wrote the test this way. It is just some packages that could cause this problem (e.g the one in https://github.com/NuGet/Home/issues/4207, and even that one won't repro anymore because NuGet fixed the underlying bug in https://github.com/NuGet/NuGet.Client/pull/1103). But if you can construct a perverse package that contains bogus entries in the RIDfull targets after restore, then Option 1 is a good replacement for this test.

It is probably OK to temporarily skip the test since [my fix](https://github.com/dotnet/sdk/pull/836) and the [NuGet fix](https://github.com/NuGet/NuGet.Client/pull/1103) make this  scenario very unlikely.